src/config/app.spec.ts

```diff
@@ -47,7 +56,41 @@ describe('config/app', () => {
                 phpIni                        : [ 'error_reporting=E_ALL' ],
                 ignorePhpPlatformRequirements : {},
                 additionalComposerArguments   : [],
+                backwardCompatibilityCheck    : false,
+                targetReference               : null,
+            });
+        });
+
+        it('should detect GITHUB_BASE_REF', () => {
+            const environment = process.env;
+
+            configDotenv({path: `${roaveBackwardCompatibilityPath}/test.env`});
+
+            expect(createConfig(
+                {
+                    codeChecks : true,
+                    docLinting : true,
+                },
+                `${roaveBackwardCompatibilityPath}/composer.json`,
+                `${roaveBackwardCompatibilityPath}/composer.lock`,
+                `${roaveBackwardCompatibilityPath}/.laminas-ci.json`
+            )).toEqual({
+                codeChecks                    : true,
+                docLinting                    : true,
+                versions                      : [],
+                stablePhpVersion              : '7.4',
+                minimumPhpVersion             : '7.4',
+                latestPhpVersion              : '7.4',
+                lockedDependenciesExists      : false,
+                phpExtensions                 : [],
+                phpIni                        : [],
+                ignorePhpPlatformRequirements : {},
+                additionalComposerArguments   : [],
+                backwardCompatibilityCheck    : true,
+                targetReference               : '1111222233334444aaaabbbbccccdddd',
```

Comment:
More than the "target" reference, this is a "base" reference.

Perhaps a more clear name would improve things a bit

Reply:
I can rename it to `baseReference` if thats more clear?

Reply:
Works for me!